src/repeating_source.rs

```diff
@@ -0,0 +1,116 @@
+use std::time::Duration;
+
+use rodio::source::Buffered;
+use rodio::Sample;
+use rodio::Source;
+
+
+/// Internal function that builds a `RepeatCount` object.
+pub fn repeat_with_count<I>(input: I,count:i32) -> RepeatCount<I>
+where
+	I: Source,
+	I::Item: Sample,
+{
+	let input = input.buffered();
+	RepeatCount {
+		inner: input.clone(),
+		next: input,
+		count: count,
+		count_remaining:count
+	}
+}
+
+/// A source that repeats the given source.
+pub struct RepeatCount<I>
+where
+	I: Source,
+	I::Item: Sample,
+{
+	inner: Buffered<I>,
+	next: Buffered<I>,
+	count: i32,
+	count_remaining: i32
+}
+
+impl<I> Iterator for RepeatCount<I>
+where
+	I: Source,
+	I::Item: Sample,
+{
+	type Item = <I as Iterator>::Item;
+
+	#[inline]
+	fn next(&mut self) -> Option<<I as Iterator>::Item> {
```

Comment:
```suggestion
	fn next(&mut self) -> Option<Self::Item> {
```
I believe this is easier to read, but you can leave it as it is if you want.